paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java

```diff
@@ -310,6 +310,27 @@ private Map<String, String> convertToProperties(Database database) {
         return properties;
     }
 
+    public void createPartition(Identifier identifier, Map<String, String> partitionSpec)
+            throws TableNotExistException {
+        TableSchema tableSchema = getDataTableSchema(identifier);
+        if (!tableSchema.partitionKeys().isEmpty()
+                && new CoreOptions(tableSchema.options()).partitionedTableInMetastore()) {
+            try {
```

Comment:
Add a warn log if tableSchema.partitionKeys().isEmpty() or tableSchema.options()).partitionedTableInMetastore() is false?